app/models/piece.rb

```diff
@@ -1,2 +1,45 @@
 class Piece < ApplicationRecord
+	
+	Pieces.each do |piece|
+		if piece.y_position == start_y && piece.x_position.between?(start_x, dest_x)
+				return false
+			end
+		end
+		return true
+	end
+
```

Comment:
@Kevlau87 it looks that start_y is not defined before it is used. Can you double check?

Reply:
@Kevlau87 does this need to be within a function?

Reply:
it is in a function and is being defined in the second line

Reply:
this also looks like an outdated commit